lib/route_translator/extensions/route_set.rb

```diff
@@ -26,9 +26,9 @@ def add_localized_route(mapping, path_ast, name, anchor, scope, path, controller
 
       def translate_mapping(locale, route_set, translated_options, translated_path_ast, scope, controller, default_action, to, formatted, via, translated_options_constraints, anchor)
         scope_params = {
-          blocks:      scope[:blocks] || [],
+          blocks:      (scope[:blocks] || []).dup,
           constraints: scope[:constraints] || {},
-          defaults:    (scope[:defaults] || {}).dup,
+          defaults:    scope[:defaults] || {},
```

Comment:
Could you please check that `dup` is done by Rails on all the versions of Rails supported?